Sources/Transform.swift

```diff
@@ -40,3 +40,76 @@ public enum Transform {
     }
 
 }
+
+#if os(iOS) || os(tvOS)
+    extension Transform: EmbeddedTransformation {
+
+        internal enum Keys {
+
+            static let name = "name"
+            static let locale = "locale"
+
+        }
+
+        internal enum Name: String {
+
+            case lowercase
+            case uppercase
+            case capitalized
+            case custom
+
+        }
+
+        internal var name: Name {
+            switch self {
+            case .lowercase, .lowercaseWithLocale: return Name.lowercase
```

Comment:
would like to take advantage of swift's type inference: 
```suggestion 
            case .lowercase, .lowercaseWithLocale: return .lowercase
```